src/formats/cif.cpp

```diff
@@ -0,0 +1,205 @@
+
+        assert(idx < total_rows);
+
+        while (idx < total_rows) {
+            if (this->is_row(table[idx])) {
+                return idx;
+            }
+            ++idx;
+        }
+
+        fmt::print(stderr, "In MCRA::find_row(), unable to find Atom(1, {}, {}, {}, {})\n"
+                    "Looping through entire table to double check...",
+                    this->_chain, this->_res_num, this->_residue, this->_atom); 
+
+        idx = 0;
+        for (const auto site : table) {
+            if (this->is_row(site)) {
+                return idx;
+            }
+            ++idx;
+        }
+        return -1;
+    }
+
+    friend std::ostream &operator<<(std::ostream &os, const MCRA &mcra) {
+        os << "Atom(" << mcra._model << " " << mcra._chain << " " << mcra._res_num << " [" << mcra._residue << "] " << mcra._atom << ")";
+        return os;
+    }
+
+private:
+    bool is_row(const gemmi::cif::Table::Row &row) const {
+        int model = std::stoul(row[11]);
+        const std::string &chain = row[1];
+        const std::string &residue = row[4];
+        const std::string &res_num = row[2];
+        const std::string atom = row[5][0] != '"' ? row[5] : row[5].substr(1, row[5].size() - 2);
+
+        if (_model == model) {
+            if (_chain == chain) {
+                if (_res_num == res_num && _residue == residue) {
+                    if (_atom == atom) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    const int _model;
+    const std::string _chain, _residue, _res_num, _atom;
+};
+
+
+void CIF::write_cif_block(std::ostream &out,
+                          gemmi::cif::Table &table, 
+                          std::vector<std::string> &p_charge, 
+                          std::vector<std::string> &vdw_radii) {
+
+    auto &loop = *table.get_loop();
+
+    unsigned long orig_tag_size = loop.tags.size();
+    unsigned long new_tag_size = orig_tag_size + 2;
+
+    // Creates a new table full of empty strings with the correct number of dimensions
+    // Outside vector size is the # of columns, inside vector size is the # of rows.
+    std::vector<std::vector<std::string>> newCols(new_tag_size, {loop.length(), {"Empty"}});
+
+    // Copies data from original columns to their respecitve column in the new table filled with empty strings.
+    // Leaving only the new appended columns as empty strings
+    for (unsigned int i = 0; i != orig_tag_size; ++i) {
+        auto iCol = table.bloc.find_loop(loop.tags[i]);
+        std::copy(iCol.begin(), iCol.end(), newCols[i].begin());
+    }
+
+    newCols[new_tag_size - 2] = std::move(p_charge);
+    newCols[new_tag_size - 1] = std::move(vdw_radii);
+
+    // TODO get this info into the column name
+    fs::path params{config::par_file};
+    std::cout << "Method name: " << config::method_name << " Parameter Name: " << params.stem() << std::endl;
```

Comment:
This is already a part of the output. Also when no parameters are specified, parameters are selected automatically but params.stem() returns nothing.